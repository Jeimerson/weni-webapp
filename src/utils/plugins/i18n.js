import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  en: en,
  'pt-br': pt_br,
  'pt-BR': pt_br,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
