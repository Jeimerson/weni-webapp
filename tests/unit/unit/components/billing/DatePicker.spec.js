import { shallowMount, createLocalVue } from '@vue/test-utils';
import DatePicker from '@/components/billing/DatePicker.vue';
import i18n from '@/utils/plugins/i18n';

import Vuex from 'vuex';
import { org } from '../../../__mocks__';
const localVue = createLocalVue();
localVue.use(Vuex);

describe('DatePicker.vue', () => {
  let wrapper;

  let actions;
  let store;
  let getters;

  beforeEach(() => {
    actions = {};
    getters = {
      currentOrg() {
        return org;
      },
    };
    store = new Vuex.Store({
      actions,
      getters,
    });
    wrapper = shallowMount(DatePicker, {
      localVue,
      i18n,
      propsData: {
        buttonAction: jest.fn(),
      },
      store,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        unnnicIconSvg: true,
        unnnicToolTip: true,
        unnnicButton: true,
        unnnicSwitch: true,
        unnnicInput: true,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
