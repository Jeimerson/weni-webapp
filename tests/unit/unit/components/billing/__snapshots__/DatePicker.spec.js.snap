// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker.vue renders a snapshot 1`] = `
<div class="container" buttonaction="function mockConstructor() {
        return fn.apply(this, arguments);
      }"><span class="label"></span>
  <div class="dropdown">
    <unnnicinput-stub size="sm" icon-left="notes-1" readonly="" value="â€”" class="input"></unnnicinput-stub>
    <div class="dropdown-data">
      <!---->
    </div>
  </div>
</div>
`;
